gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -671,12 +674,29 @@ public FSDataOutputStream createNonRecursive(
       throws IOException {
     URI gcsPath = getGcsPath(checkNotNull(hadoopPath, "hadoopPath must not be null"));
     URI parentGcsPath = UriPaths.getParentPath(gcsPath);
+    DurationTracker get_tracker = getDurationTrackerFactory().trackDuration(GHFSStatistic.ACTION_HTTP_GET_REQUEST.getSymbol());
+    DurationTracker head_tracker = getDurationTrackerFactory().trackDuration(GHFSStatistic.ACTION_HTTP_HEAD_REQUEST.getSymbol());
+
     if (!getGcsFs().getFileInfo(parentGcsPath).exists()) {
       throw new FileNotFoundException(
           String.format(
               "Can not create '%s' file, because parent folder does not exist: %s",
               gcsPath, parentGcsPath));
     }
+    if (this.getGcsFs().getGcs().getStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES) > 0) {
```

Comment:
This is very manual process, can we use delegation instead? I.e. increment only `GoogleCloudStorageStatistics` and use it to implement `GHFSStatistics`? Something like this:

```java
class GHFSStatistics {
   private GoogleCloudStorageStatistics delegate = ...;
  ...
  public long getStatistics(String key) {
    return delegate.getStatistics(key);
  }
}
```

Reply:
Updated